File: src/main/java/net/hectus/neobb/turn/legacy_game/structure/LTNetherPortal.java
package net.hectus.neobb.turn.legacy_game.structure;

import net.hectus.neobb.NeoBB;
import net.hectus.neobb.game.Game;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.DummyTurn;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.function.AttackFunction;
import net.hectus.neobb.turn.default_game.other.OtherTurn;
import net.hectus.neobb.util.Modifiers;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.Material;
import org.bukkit.block.BlockState;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class LTNetherPortal extends OtherTurn<List<BlockState>> implements HotClazz, AttackFunction, DummyTurn {
    public LTNetherPortal(NeoPlayer player) { super(player); }
    public LTNetherPortal(List<BlockState> data, Location location, NeoPlayer player) { super(data, location, player); }

    @Override
    public int cost() {
        return 3;
    }

    @Override
    public List<ItemStack> items() {
        return List.of(new ItemStack(Material.OBSIDIAN, 14));
    }

    public static void await(@NotNull Game game) {
        game.addModifier(Modifiers.G_LEGACY_PORTAL_AWAIT);
        Bukkit.getScheduler().runTaskLater(NeoBB.PLUGIN, () -> {
            if (game.hasModifier(Modifiers.G_LEGACY_PORTAL_AWAIT)) {
                game.removeModifier(Modifiers.G_LEGACY_PORTAL_AWAIT);
                game.draw(false);
            }
        }, 100); // 5 Seconds
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/legacy_game/structure/LTNetherPortal.kt
package net.hectus.neobb.modes.turn.legacy_game.structure

import com.marcpg.libpg.data.time.Time
import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.game.Game
import net.hectus.neobb.modes.turn.ComboTurn
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.AttackFunction
import net.hectus.neobb.modes.turn.default_game.other.OtherTurn
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.util.Modifiers
import net.hectus.neobb.util.bukkitRunLater
import org.bukkit.block.BlockState

class LTNetherPortal(data: List<BlockState>?, cord: Cord?, player: NeoPlayer?) : OtherTurn<List<BlockState>>(data, cord, player), ComboTurn, AttackFunction, HotClazz {
    companion object {
        fun await(game: Game) {
            game.addModifier(Modifiers.Game.Legacy.PORTAL_AWAIT)
            bukkitRunLater(Time(5)) {
                if (game.hasModifier(Modifiers.Game.Legacy.PORTAL_AWAIT)) {
                    game.removeModifier(Modifiers.Game.Legacy.PORTAL_AWAIT)
                    game.draw()
                }
            }
        }
    }

    override val cost: Int = 3
}
